threads/timeminer.py

```diff
@@ -122,84 +116,36 @@ def ping_worker(self):
 
     def tick_worker(self):
         while True:
-            # added_ping acts as a switch between "pingmode" and "tickmode"
-            if self.networker.ready and self.added_ping:
-                # Always construct tick in the following order:
-                # 1) Init 2) Mine+nonce 3) Add signature
-                # This is because the order of nonce and sig creation matters
-                cycle_time = config['cycle_time']
-                cycle_multiplier = config['cycle_time_multiplier']
-
-                # Dynamic adjusting of sleeping time to match network lockstep
-                prev_tick_ts = median_ts(self.clockchain.latest_selected_tick())
-                desired_ts = prev_tick_ts + cycle_multiplier*cycle_time
-
-                wait_time = desired_ts - utcnow()
-
-                logger.debug("Median ts: " + str(prev_tick_ts) + " min ts: "
-                             + str(desired_ts) + " curr ts: " + str(utcnow()))
-
-                overshoot = 0
-
-                if wait_time < 0:
-                    if self.clockchain.current_height() != 0:  # If init, ignore
-                        overshoot = -wait_time
-                    logger.debug("Overshoot of " + str(int(overshoot)) + "s")
-                    wait_time = 0
+            time.sleep(5)
+            if self.networker.ready and not self.networker.stage == 'select':
 
-                logger.debug("Adjusted sleeping time: " + str(int(wait_time)))
-                time.sleep(wait_time)  # Adjusting to follow network timing
+                if len(list(self.clockchain.ping_pool.values())) == 0:
+                    logger.info("(tick_worker) No pings, waiting")
+                    continue
 
+                logger.info("(tick_worker) Pingpool not empty, building tick")
                 logger.debug("Tick stage--------------------------------------")
-                start = time.time()  # Start and end time used to adjust sleep
-
-                self.networker.stage = "tick"
 
                 self.generate_and_process_tick()
```

Comment:
to add: verbose=False / True:

In order to avoid confusing output in sync mechanism